[WIP] V2021.10 nvec paz00

kwizart opened the pull request:
This is a rebase of my tree from 2021.10 nvec paz00 driver.

Known not to work is EFI boot on tegra20 (works for tegra124).
please review.

marvintwentyfour committed:
ARM: tegra: add nvec driver

Adopted version of nvec driver from linux kernel with
keyboard support only.

Signed-off-by: Andrey Danin <danindrey@mail.ru>
[squashed, forward ported, and cleaned up]
Signed-off-by: Marc Dietrich <marvin24@gmx.de>

marvintwentyfour committed:
ARM: tegra: add nvec keyboard driver

Adopted version of nvec keyboard driver from linux kernel.

Signed-off-by: Andrey Danin <danindrey@mail.ru>
[squashed, forward ported, and cleaned up]
Signed-off-by: Marc Dietrich <marvin24@gmx.de>

marvintwentyfour committed:
ARM: tegra: paz00: enable nvec keyboard support

Signed-off-by: Andrey Danin <danindrey@mail.ru>
[squashed, forward ported, and cleaned up]
Signed-off-by: Marc Dietrich <marvin24@gmx.de>

kwizart committed:
i2c: tegra_nvec: Use dev_read_addr_ptr/u32_default

Signed-off-by: Nicolas Chauvet <kwizart@gmail.com>

ziswiler committed:
tegra: lcd: video: integrate display driver for t30

On popular request make the display driver from T20 work on T30 as
well. Turned out to be quite straight forward. However a few notes
about some things encountered during porting: Of course the T30 device
tree was completely missing host1x as well as PWM support but it turns
out this can simply be copied from T20. The only trouble compiling the
Tegra video driver for T30 had to do with some hard-coded PWM pin
muxing for T20 which is quite ugly anyway. On T30 this gets handled by
a board specific complete pin muxing table. The older Chromium U-Boot
2011.06 which to my knowledge was the only prior attempt at enabling a
display driver for T30 for whatever reason got some clocking stuff
mixed up. Turns out at least for a single display controller T20 and
T30 can be clocked quite similar. Enjoy.

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

digetx committed:
Tegra: Support Tegra20 QEMU board

Add device-tree and config for Tegra20 QEMU.

Link: https://github.com/grate-driver/tegra2_qemu
Signed-off-by: Dmitry Osipenko <digetx@gmail.com>

kwizart commented on drivers/i2c/tegra_nvec.c:
```diff
@@ -410,7 +411,7 @@ static int nvec_probe(struct udevice *dev)
 	struct nvec_t *nvec = dev_get_priv(dev);
 	const void *blob = gd->fdt_blob;
-	int node = dev->node.of_offset;
+	int node = dev->node_.of_offset;
```
This should probably be deleted (can be squashed with the later patch).

kwizart commented on drivers/i2c/tegra_nvec.c:
```diff
@@ -425,7 +421,7 @@ static int nvec_probe(struct udevice *dev)
 	debug("NVEC initialization...\n");
 
-	if (nvec_decode_fdt(blob, node, nvec_data)) {
+	if (nvec_decode_fdt(nvec_data) {
```
Missing bracket,  will fix...

kwizart commented:
It fails to build from source, I will revisit in the next days. I need to recover a known good and tested version...

clamor-s commented:
> It fails to build from source, I will revisit in the next days. I need to recover a known good and tested version...

Sources were updated. May you rebase your PR please?

kwizart commented:
Sure, I hope to update this over the week-end.

clamor-s committed:
asus: grouper: initial support

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

jonasschwoebel committed:
input: gpio-keys: create simple driver for handling gpio keys

Signed-off-by: Jonas Schwöbel <jonasschwoebel@yahoo.de>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add transformer t20 board support

Board derives from Ventana board

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add transformer t30 board support

Board derives from Cardhu board

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF101 support

Tested-by: Antoni Aloy Torrens <aaloytorrens@gmail.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF101G support

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add SL101 support

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF201 support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF300T support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF300TG support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF300TL support

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF600T support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add TF700T support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
asus: add P1801-T support

Tested-by: Andreas Westman Dorcsak <hedmoo@yahoo.com>
Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s committed:
part_efi: support backup gpt use

This patch adds support for correct work with backup gpt
without excessive log spam.

Signed-off-by: Svyatoslav Ryhel <clamor95@gmail.com>

clamor-s commented:
> Sure, I hope to update this over the week-end.

Rebase your changes please.

kwizart commented:
The nvec fork for paz00 is currently maintained at 
https://gitlab.com/ac100/u-boot-ac100-exp/-/merge_requests/2
I'm seeking review from @marvintwentyfour in a first step.

I've rebased from grate until then.

kwizart commented:
Closing as I've submitted a new PR with a new base branch
https://github.com/grate-driver/u-boot/pull/2
